lib/addons/src/make-decorator.ts

```diff
@@ -1,23 +1,41 @@
 import deprecate from 'util-deprecate';
 
-// Create a decorator that can be used both in the (deprecated) old "hoc" style:
-//   .add('story', decorator(options)(() => <Story />));
-//
-// And in the new, "parameterized" style:
-//   .addDecorator(decorator)
-//   .add('story', () => <Story />, { name: { parameters } });
-//
-// *And* in the older, but not deprecated, "pass options to decorator" style:
-//  .addDecorator(decorator(options))
-
-export const makeDecorator = ({
+export interface StoryContext {
+  story: string;
+  kind: string;
+}
+
+export interface WrapperSettings {
+  options: object;
+  parameters: any;
+}
+
+export type StoryGetter = (context: StoryContext) => any;
+
+export type StoryWrapper = (
+  getStory: StoryGetter,
+  context: StoryContext,
+  settings: WrapperSettings
+) => any;
+
+type MakeDecoratorResult = (...args: any) => any;
```

Comment:
I believe this line wasn't correctly typed actually... It's not simply saying there is no type (`any`) but TSLint should point out an error saying args must be an array, so it should be `any[]`.